instrumentation/spring/spring-boot-autoconfigure/src/main/java/io/opentelemetry/instrumentation/spring/autoconfigure/propagators/CompositeTextMapPropagatorFactory.java

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.instrumentation.spring.autoconfigure.propagators;
+
+import io.opentelemetry.api.baggage.propagation.W3CBaggagePropagator;
+import io.opentelemetry.api.trace.propagation.W3CTraceContextPropagator;
+import io.opentelemetry.context.propagation.TextMapPropagator;
+import io.opentelemetry.extension.aws.AwsXrayPropagator;
+import io.opentelemetry.extension.trace.propagation.B3Propagator;
+import io.opentelemetry.extension.trace.propagation.JaegerPropagator;
+import io.opentelemetry.extension.trace.propagation.OtTracePropagator;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import org.springframework.beans.factory.BeanFactory;
+import org.springframework.util.ClassUtils;
+
+/** Factory of composite {@link TextMapPropagator}. Defaults to W3C and BAGGAGE. */
+public final class CompositeTextMapPropagatorFactory {
+
+  static TextMapPropagator getCompositeTextMapPropagator(
+      BeanFactory beanFactory, List<PropagationType> types) {
+
+    Set<TextMapPropagator> propagators = new HashSet<>();
+
+    for (PropagationType type : types) {
+      switch (type) {
+        case b3:
+          if (isOnClasspath("io.opentelemetry.extension.trace.propagation.B3Propagator")) {
+            propagators.add(
+                beanFactory
+                    .getBeanProvider(B3Propagator.class)
+                    .getIfAvailable(B3Propagator::injectingSingleHeader));
```

Comment:
Does this mean that you have to have a bean of type `B3Propagator` in your application?
Why not just
```suggestion
            propagators.add(B3Propagator.injectingSingleHeader());
```
?

Reply:
The current approach uses the `B3Propagator` bean if it's present, but if it's not it creates an instance in `getIfAvailable` (similarly as in https://github.com/spring-cloud-incubator/spring-cloud-sleuth-otel/blob/main/spring-cloud-sleuth-otel/src/main/java/org/springframework/cloud/sleuth/otel/propagation/CompositeTextMapPropagator.java)

Reply:
Oh, I see - that's pretty cool 👍